mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py

```diff
@@ -0,0 +1,354 @@
+        for gate_center in gate_centers:
+            distances.append((gate_center[0] - pinger_pos[0]) ** 2 + (gate_center[1] - pinger_pos[1]) ** 2)
+        argmin = np.argmin(np.array(distances))
+
+        return argmin
+
+    @util.cancellableInlineCallbacks
+    def get_scan_point(self, gates_line, center_gate):
+        traversal_vector = self.perpendicular(gates_line)
+
+        boat_pose = yield self.tx_pose
+        boat_pose = boat_pose[0]
+
+        recenter_distance = 15
+
+        recenter_points = [(center_gate[0] + traversal_vector[0] * recenter_distance,
+                            center_gate[1] + traversal_vector[1] * recenter_distance),
+                           (center_gate[0] + traversal_vector[0] * -recenter_distance,
+                            center_gate[1] + traversal_vector[1] * -recenter_distance)]
+
+        if (recenter_points[0][0] - boat_pose[0]) ** 2 + (recenter_points[0][1] - boat_pose[1]) ** 2 > \
+                (recenter_points[1][0] - boat_pose[0]) ** 2 + \
+                (recenter_points[1][1] - boat_pose[1]) ** 2:
+            recenter_points = [recenter_points[1], recenter_points[0]]
+
+        recenter_points_np = []
+
+        for goal_point in recenter_points:
+            point = np.array(goal_point)
+            point = np.append(point, [0])
+            recenter_points_np.append(point)
+
+        defer.returnValue(recenter_points_np)
+
+    @util.cancellableInlineCallbacks
+    def get_traversal_points(self, gates_line, pinger_gate, center_gate):
+        traversal_vector = self.perpendicular(gates_line)
+
+        boat_pose = yield self.tx_pose
+        boat_pose = boat_pose[0]
+
+        traversal_distance = 7
+        traversal_points = [(pinger_gate[0] + traversal_vector[0] * traversal_distance,
+                             pinger_gate[1] + traversal_vector[1] * traversal_distance),
+                            (pinger_gate[0] + traversal_vector[0] * -traversal_distance,
+                             pinger_gate[1] + traversal_vector[1] * -traversal_distance)]
+
+        if (traversal_points[0][0] - boat_pose[0]) ** 2 + (traversal_points[0][1] - boat_pose[1]) ** 2 > (
+                traversal_points[1][0] - boat_pose[0]) ** 2 + \
+                (traversal_points[1][1] - boat_pose[1]) ** 2:
+            traversal_points = [traversal_points[1], traversal_points[0]]
+
+        traversal_points_np = []
+
+        for goal_point in traversal_points:
+            point = np.array(goal_point)
+            point = np.append(point, [0])
+            traversal_points_np.append(point)
+
+        defer.returnValue(traversal_points_np)
+
+    @util.cancellableInlineCallbacks
+    def heading_cb(self, p_message):
+        if self.enablePingerTracking:
+            try:
+                transformed_vec = self.tfBuffer.transform(p_message,
+                        self.global_frame, rospy.Duration(2))
+                transformed_origin = self.tfBuffer.lookup_transform(self.global_frame,
```

Comment:
Hmmm would be surprised if this didn't require a yield... be sure to test mission in sim to check little things like this